core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,696 @@ def update_question_suggestion(
+            user_id
+        )
+    )
+    return [
+        _create_question_contribution_stats_from_models(model)
+        for model in question_contribution_stats_models
+    ]
+
+
+def _get_all_question_review_stats(
+    user_id: str) -> List[suggestion_registry.QuestionReviewStats]:
+    """Gets all QuestionReviewStatsModels corresponding to the supplied
+    user and converts them to their corresponding domain objects.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        list(QuestionReviewStats). QuestionReviewStats domain objects
+        corresponding to the supplied user.
+    """
+    question_review_stats_models = (
+        suggestion_models.QuestionReviewStatsModel.get_all_by_user_id(
+            user_id
+        )
+    )
+    return [
+        _create_question_review_stats_from_models(model)
+        for model in question_review_stats_models
+    ]
+
+
+def get_all_contributor_stats(
+    user_id: str) -> suggestion_registry.ContributorStatsSummary:
+    """Gets ContributorStatsSummary corresponding to the supplied user.
+
+    Args:
+        user_id: str. User ID.
+
+    Returns:
+        ContributorStatsSummary. ContributorStatsSummary domain objects
+        corresponding to the supplied user.
+    """
+    translation_contribution_stats = (get_all_translation_contribution_stats(
+        user_id))
+    translation_review_stats = _get_all_translation_review_stats(user_id)
+    question_contribution_stats = _get_all_question_contribution_stats(user_id)
+    question_review_stats = _get_all_question_review_stats(user_id)
+
+    return suggestion_registry.ContributorStatsSummary(
+        user_id,
+        translation_contribution_stats,
+        question_contribution_stats,
+        translation_review_stats,
+        question_review_stats)
+
+
+def _update_translation_contribution_stats_models(
+    translation_contribution_stats: List[
+        suggestion_registry.TranslationContributionStats]) -> None:
+    """Updates TranslationContributionStatsModel models for given translation
+    contribution stats.
+
+    Args:
+        translation_contribution_stats: list(TranslationContributionStats).
+            A list of TranslationContributionStats domain objects.
+    """
+    stats_ids = []
+
+    for stat in translation_contribution_stats:
+        # Ruling out the possibility of None for mypy type checking.
+        assert stat is not None
+        stats_ids.append(
+            suggestion_models.TranslationContributionStatsModel.generate_id(stat.language_code, stat.contributor_user_id, stat.topic_id)) # type: ignore[arg-type]
+
+    stats_models_to_update = (
+        suggestion_models.TranslationContributionStatsModel.get_multi(
+            stats_ids))
+
+    for index, stats_model in enumerate(stats_models_to_update):
+
+        # Ruling out the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = translation_contribution_stats[index]
```

Comment:
Is it guaranteed that `suggestion_models.TranslationContributionStatsModel.get_multi` will return the corresponding models in the same order? Maybe create a dict of stats model ID -> stats domain object so that we can lookup by ID. Here and below.

Reply:
Done